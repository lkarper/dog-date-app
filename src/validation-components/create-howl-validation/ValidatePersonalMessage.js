import React, { useEffect } from 'react';

const ValidatePersonalMessage = (props) => {
    const { personalMessage, personalMessageError, setPersonalMessageError } = props;

    useEffect(() => {
        if (personalMessage && personalMessage.trim().length === 0) {
            setPersonalMessageError('If you include a personal message, it must contain more than spaces.')
        } else {
            setPersonalMessageError(null);
        }

    }, [personalMessage, setPersonalMessageError]);

    if (personalMessageError) {
        return <p id='personal-message-validator'>{personalMessageError} (Limited to 2000 characters.)</p>
    }

    return <p id='personal-message-validator'>Personal message optional. (Limited to 2000 characters.)</p>
}


export default ValidatePersonalMessage;
