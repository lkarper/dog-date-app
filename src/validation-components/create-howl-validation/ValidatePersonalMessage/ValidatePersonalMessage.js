import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidatePersonalMessage = (props) => {
    const { 
        personalMessage, 
        personalMessageError, 
        setPersonalMessageError, 
    } = props;

    useEffect(() => {
        if (personalMessage && personalMessage.trim().length === 0) {
            setPersonalMessageError('If you include a personal message, it must contain more than spaces.')
        } else {
            setPersonalMessageError('');
        }
    }, [personalMessage, setPersonalMessageError]);

    if (personalMessageError) {
        return (
            <p 
                className='ValidatePersonalMessage__validator error'
                id='personal-message-validator'
            >
                {personalMessageError} (Limited to 2000 characters.)
            </p>
        );
    }

    return (
        <p 
            className='ValidatePersonalMessage__validator'
            id='personal-message-validator'
        >
            Personal message optional. (Limited to 2000 characters.)
        </p>
    );
}

ValidatePersonalMessage.defaultProps = {
    personalMessage: '',
    personalMessageError: '',
    setPersonalMessageError: () => {},
};

ValidatePersonalMessage.propTypes = {
    personalMessage: PropTypes.string.isRequired,
    personalMessageError: PropTypes.string.isRequired,
    setPersonalMessageError: PropTypes.func.isRequired,
};

export default ValidatePersonalMessage;
