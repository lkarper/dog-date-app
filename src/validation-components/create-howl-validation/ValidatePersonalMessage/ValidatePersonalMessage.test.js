import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidatePersonalMessage from './ValidatePersonalMessage';

describe('ValidatePersonalMessage component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidatePersonalMessage />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidatePersonalMessage />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
