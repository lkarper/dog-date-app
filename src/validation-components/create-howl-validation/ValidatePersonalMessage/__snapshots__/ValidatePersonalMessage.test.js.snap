// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidatePersonalMessage component renders the UI as expected 1`] = `
<ValidatePersonalMessage
  personalMessage=""
  personalMessageError=""
  setPersonalMessageError={[Function]}
>
  <p
    className="ValidatePersonalMessage__validator"
    id="personal-message-validator"
  >
    Personal message optional. (Limited to 2000 characters.)
  </p>
</ValidatePersonalMessage>
`;
